frontend/src/metabase/admin/tasks/containers/JobTriggersModal.jsx

```diff
@@ -53,40 +51,25 @@ const renderTriggersTable = triggers => {
   );
 };
 
-class JobTriggersModal extends Component {
-  state = {
-    triggers: null,
-    error: null,
-  };
-
-  async componentDidMount() {
-    try {
-      const { jobKey } = this.props.params;
-      const jobs = jobKey && (await this.props.fetchJobInfo()).payload.jobs;
-      const job = jobs && _.findWhere(jobs, { key: jobKey });
-      const triggers = (job && job.triggers) || [];
-
-      this.setState({ triggers, error: null });
-    } catch (error) {
-      this.setState({ error });
-    }
-  }
-
-  render() {
-    const {
-      params: { jobKey },
-      goBack,
-    } = this.props;
-    const { triggers, error } = this.state;
+export const JobTriggersModal = props => {
+  const dispatch = useDispatch();
+  const { data, error, isFetching } = useGetTasksInfoQuery();
 
-    return (
-      <ModalContent title={t`Triggers for ${jobKey}`} onClose={goBack}>
-        <LoadingAndErrorWrapper loading={!triggers} error={error}>
-          {() => renderTriggersTable(triggers)}
-        </LoadingAndErrorWrapper>
-      </ModalContent>
-    );
-  }
-}
+  const { jobKey } = props.params;
+  const jobs = jobKey && data?.jobs;
+  const job = jobs && _.findWhere(jobs, { key: jobKey });
 
-export default connect(null, { fetchJobInfo, goBack })(JobTriggersModal);
+  return (
+    <ModalContent
+      title={t`Triggers for ${jobKey}`}
+      onClose={() => dispatch(goBack())}
+    >
+      <LoadingAndErrorWrapper
+        loading={isFetching && !job?.triggers}
```

Comment:
I don't think wee need `!job?.triggers` check now that we have proper `isFetching` state.

```suggestion
        loading={isFetching}
```